allennlp/semparse/contexts/table_question_knowledge_graph.py

```diff
@@ -61,6 +61,82 @@
         **MONTH_NUMBERS,
         }
 
+
+class ColumnTyper():
+    """
+        Type table columns as <string> | <number> | <date> 
+    """
+    pass
+
+
+
+class QuestionEntityExtractor():
+    """
+    A Barebones implementation similar to https://github.com/crazydonkey200/neural-symbolic-machines/blob/master/table/wtq/preprocess.py
+    for extracting entities from a question given a table. 
+    """
+    def __init__(self, 
+                 stop_words_file : str,
+                 table_entity_text: Dict[str, str]) -> None:
+        self.table_entity_text = table_entity_text
+        with open(stop_words_file, 'r') as f:
+            stop_words_list = json.load(f)
+        self.stop_words = set(stop_words_list)
+
+
+    def _string_in_table(self, candidate_str) -> bool:
+        for entity, text in self.table_entity_text.items():
+            # entity normalized text is followed after fb:cell.
+            if entity.startswith('fb:cell.')
+            normalized_text = entity[8:]
+            if candidate_str in normalized_text: 
```

Comment:
`candidate_str` comes from `TableQuestionKnowledgeGraph._normalize_string` that does not replace spaces with underscores, correct? But `normalized_text` here has underscores. You'll have to make sure they are comparable.